flink-staging/flink-table/src/main/scala/org/apache/flink/api/table/Table.scala

```diff
@@ -243,5 +243,29 @@ case class Table(private[flink] val operation: PlanNode) {
     this.copy(operation = Join(operation, right.operation))
   }
 
+  /**
+   * Union two[[Table]]s. Similar to an SQL UNION ALL. The fields of the two union operations
+   * must fully overlap.
+   *
+   * Example:
+   *
+   * {{{
+   *   left.unionAll(right)
+   * }}}
+   */
+  def unionAll(right: Table): Table = {
+    val leftInputNames = operation.outputFields.map(_._1).toSet
+    val rightInputNames = right.operation.outputFields.map(_._1).toSet
+    if (!leftInputNames.equals(rightInputNames)) {
```

Comment:
I think checking the output names is not enough, the types must also be equal otherwise we might get undesired behavior. You should remove the `map`s.